src/promptflow/tests/sdk_cli_azure_test/e2etests/test_flow_serve.py

```diff
@@ -5,14 +5,108 @@
+    response = flow_serving_client_with_prt_config_env.post("/score", data=json.dumps({"text": "hi"}))
+    assert (
+        response.status_code == 200
+    ), f"Response code indicates error {response.status_code} - {response.data.decode()}"
+    assert "output_prompt" in json.loads(response.data.decode())
+    response = flow_serving_client_with_prt_config_env.get("/")
+    assert b"Welcome to promptflow app" in response.data
+
+
+@pytest.mark.skipif(condition=not is_live(), reason="serving tests, only run in live mode.")
+@pytest.mark.usefixtures("flow_serving_client_with_connection_provider_env")
+@pytest.mark.e2etest
+def test_azureml_serving_api_with_conn_provider_env(flow_serving_client_with_connection_provider_env):
+    response = flow_serving_client_with_connection_provider_env.get("/health")
+    assert b'{"status":"Healthy","version":"0.0.1"}' in response.data
+    response = flow_serving_client_with_connection_provider_env.post("/score", data=json.dumps({"text": "hi"}))
+    assert (
+        response.status_code == 200
+    ), f"Response code indicates error {response.status_code} - {response.data.decode()}"
+    assert "output_prompt" in json.loads(response.data.decode())
+    response = flow_serving_client_with_connection_provider_env.get("/")
+    assert b"Welcome to promptflow app" in response.data
+
+
+@pytest.mark.skipif(condition=not is_live(), reason="serving tests, only run in live mode.")
+@pytest.mark.usefixtures("flow_serving_client_with_connection_provider_env")
+@pytest.mark.e2etest
+def test_azureml_serving_api_with_aml_resource_id_env(flow_serving_client_with_aml_resource_id_env):
+    response = flow_serving_client_with_aml_resource_id_env.get("/health")
+    assert b'{"status":"Healthy","version":"0.0.1"}' in response.data
+    response = flow_serving_client_with_aml_resource_id_env.post("/score", data=json.dumps({"text": "hi"}))
+    assert (
+        response.status_code == 200
+    ), f"Response code indicates error {response.status_code} - {response.data.decode()}"
+    assert "output_prompt" in json.loads(response.data.decode())
+
+
+@pytest.mark.skipif(condition=not is_live(), reason="serving tests, only run in live mode.")
+@pytest.mark.usefixtures("serving_client_with_connection_name_override")
+@pytest.mark.e2etest
+def test_azureml_serving_api_with_connection_name_override(serving_client_with_connection_name_override):
+    response = serving_client_with_connection_name_override.get("/health")
+    assert b'{"status":"Healthy","version":"0.0.1"}' in response.data
+
+    response = serving_client_with_connection_name_override.post("/score", data=json.dumps({"text": testdata}))
+    assert (
+        response.status_code == 200
+    ), f"Response code indicates error {response.status_code} - {response.data.decode()}"
+    assert "output_prompt" in json.loads(response.data.decode())
+
+
+@pytest.mark.skipif(condition=not is_live(), reason="serving tests, only run in live mode.")
```

Comment:
we better remove the restriction that this tests only run in live mode.

Reply:
i just follow the existing test,  could anyone help explain what does this live mode mean here?

Reply:
means without reply. 
You can check this: https://github.com/microsoft/promptflow/blob/main/docs/dev/replay-e2e-test.md

Reply:
most of the tests are remote connection testing, i'll remove the restriction for those which don't need remote connection.

Reply:
These tests are mainly for the new auzreml extension, which has very limited dependency on authoring part, so only run in live mode for some of the tests which depends on remote connection should be fine.